### PR TITLE
limiting storage and processing snapshot versions of projects

### PR DESCRIPTION
**Pull request by gs-gunjan**
limiting snapshot versions of projects

### CODE REVIEW

File: legend-depot-artifacts-refresh/src/main/java/org/finos/legend/depot/store/artifacts/services/ProjectVersionRefreshHandler.java [hunk 1]
```diff
@@ -82,10 +83,11 @@ public final class ProjectVersionRefreshHandler implements NotificationEventHand
     private final List<String> projectPropertiesInScope;
     private final Queue workQueue;
     private final DependencyManager dependencyManager;
+    private final int noOfSnapshotVersions;
 
 
     @Inject
-    public ProjectVersionRefreshHandler(ManageProjectsService projects, RepositoryServices repositoryServices, Queue workQueue, RefreshStatusStore store, ArtifactsFilesStore artifacts, IncludeProjectPropertiesConfiguration includePropertyConfig, DependencyManager dependencyManager)
+    public ProjectVersionRefreshHandler(ManageProjectsService projects, RepositoryServices repositoryServices, Queue workQueue, RefreshStatusStore store, ArtifactsFilesStore artifacts, IncludeProjectPropertiesConfiguration includePropertyConfig, DependencyManager dependencyManager, RetentionPolicyConfiguration retentionPolicyConfiguration)
```
Comment (elopezcastro) [inline legend-depot-artifacts-refresh/src/main/java/org/finos/legend/depot/store/artifacts/services/ProjectVersionRefreshHandler.java:90]: please dont pass the entiry configuration just the specific item u need from it..
the place  where u have put the definition for the config is not correct. move it o where it belongs and expose wht u need from it

File: legend-depot-model/src/main/java/org/finos/legend/depot/domain/project/RetentionPolicyConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+//  Copyright 2021 Goldman Sachs
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
```
Comment (elopezcastro) [inline legend-depot-model/src/main/java/org/finos/legend/depot/domain/project/RetentionPolicyConfiguration.java:6]: this does not belong in the model is specific to retention

File: legend-depot-model/src/main/java/org/finos/legend/depot/domain/project/RetentionPolicyConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+package org.finos.legend.depot.domain.project;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.validation.constraints.NotNull;
+
+public class RetentionPolicyConfiguration
+{
+    @NotNull
```
Comment (elopezcastro) [inline legend-depot-model/src/main/java/org/finos/legend/depot/domain/project/RetentionPolicyConfiguration.java:25]: can ypu specify a default if not present

File: legend-depot-artifacts-refresh/src/main/java/org/finos/legend/depot/store/artifacts/services/ProjectVersionRefreshHandler.java [hunk 4]
```diff
@@ -144,6 +147,10 @@ public List<String> validateEvent(MetadataNotification event)
         {
             errors.add(String.format("Invalid projectId [%s]. Existing project [%s] has same [%s-%s] coordinates",event.getProjectId(),projectData.get().getProjectId(),event.getGroupId(),event.getArtifactId()));
         }
+        if (projects.findSnapshotVersions(event.getGroupId(), event.getArtifactId()).size() >= noOfSnapshotVersions)
+        {
+            errors.add(String.format("Number of snapshot versions stored for project %s-%s, has reached the limit", event.getGroupId(), event.getArtifactId()));
```
Comment (elopezcastro) [inline legend-depot-artifacts-refresh/src/main/java/org/finos/legend/depot/store/artifacts/services/ProjectVersionRefreshHandler.java:152]: add the limit to the error message too

**Review by elopezcastro — CHANGES_REQUESTED** [review]

File: legend-depot-artifacts-refresh/src/main/java/org/finos/legend/depot/store/artifacts/services/ProjectVersionRefreshHandler.java (hunk 1 above)
Comment (gs-gunjan) [reply legend-depot-artifacts-refresh/src/main/java/org/finos/legend/depot/store/artifacts/services/ProjectVersionRefreshHandler.java]: now I inject only the snapshot information

File: legend-depot-model/src/main/java/org/finos/legend/depot/domain/project/RetentionPolicyConfiguration.java (hunk 3 above)
Comment (gs-gunjan) [reply legend-depot-model/src/main/java/org/finos/legend/depot/domain/project/RetentionPolicyConfiguration.java]: added this

File: legend-depot-artifacts-refresh/src/main/java/org/finos/legend/depot/store/artifacts/configurations/ArtifactsRetentionPolicyConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.ws.rs.DefaultValue;
+
+public class ArtifactsRetentionPolicyConfiguration
+{
+    @JsonProperty
+    @DefaultValue(value = "20")
+    private final int noOfSnapshotVersions;
```
Comment (elopezcastro) [inline legend-depot-artifacts-refresh/src/main/java/org/finos/legend/depot/store/artifacts/configurations/ArtifactsRetentionPolicyConfiguration.java:27]: can we call this maximunSnaphotsAllowed

**Review by elopezcastro — CHANGES_REQUESTED** [review]

**Review by elopezcastro — APPROVED** [review]

**Pull request merged**